src/agora/dobeamweights.cc

```diff
@@ -389,18 +389,32 @@ static inline void TransposeGather(size_t cur_sc_id, float* src, float*& dst,
 }
 
 void DoBeamWeights::ComputeBeams(size_t tag) {
+  //Request was generated from gen_tag_t::FrmSc
   const size_t frame_id = gen_tag_t(tag).frame_id_;
   const size_t base_sc_id = gen_tag_t(tag).sc_id_;
   const size_t frame_slot = frame_id % kFrameWnd;
   if (kDebugPrintInTask) {
     std::printf("In doZF thread %d: frame: %zu, base subcarrier: %zu\n", tid_,
                 frame_id, base_sc_id);
   }
-  const size_t num_subcarriers =
-      std::min(cfg_->BeamBlockActiveSc(), cfg_->OfdmDataNum() - base_sc_id);
+
+  //Process BeamBlockSize (or less) number of carriers
+  //cfg_->OfdmDataNum() is the total number of usable subcarriers
+  size_t num_subcarriers =
+      std::min(cfg_->BeamBlockSize(), cfg_->OfdmDataNum() - base_sc_id);
 
   // Handle each subcarrier one by one
-  for (size_t i = 0; i < num_subcarriers; i++) {
+  size_t sc_inc = 1;
+  size_t start_sc = 0;
+  if (cfg_->FreqOrthogonalPilot()) {
+    //For FreqOrthogonalPilot only process the first sc in each group
+    sc_inc = cfg_->PilotScGroupSize();
+    //Start at the 1st multiple of sc_inc
+    start_sc = sc_inc - (base_sc_id % sc_inc);
```

Comment:
I think start_sc should always be 0. No start offset is needed.
(base_sc_id % sc_inc) == 0, given that beam_block_size == pilot_sc_group_size. According to this line, start_sc == pilot_sc_group_size, i.e. offset to skip the entire group, which doesn't seem right.

Reply:
I would like to move the restriction on beam block size.   Therefore you can assume the function will allow the input of any BeamBlockSize.   I agree this math is wrong.

Process sc for PilotScGroupSize  = 0.   0, 8, 16, 24, 32, 40, 48, 56, 64, 72, 
Assume PilotScGroupSize = 8, and BeamBlockSize = 7.   Inputs will be 0, 7, 14, 21, 28, 35, 42, 49, 56.  Offsets should be 0, 1, 2,3,4,5,6,7 (where start_sc = 7, and num_subcarriers = 7, no execute) ,0

Assume PilotScGroupSize = 8, and BeamBlockSize = 9.   Inputs will be 0, 9, 18, 27, 36, 45, 54, 63, 72.  Offsets should be 0[0,8], 7[16],  6[24], 5 [32], 4 [40], 3 [48], 2 [56], 1 [64], 0 [72,80].

Reply:
```
const size_t rem = base_sc_id % cfg_->PilotScGroupSize();
if (rem == 0) {
 start_sc = 0;
else {
 start_sc = cfg_->PilotScGroupSize() - rem ;
}
```
Does this look right?

Reply:
I think it is right. The else block can be commented as "offset start sc to the start of next pilot sc group".

Reply:
Completed